src/main/java/com/minecolonies/core/colony/crafting/ItemStorageFactory.java

```diff
@@ -90,7 +91,31 @@ public ItemStorage deserialize(@NotNull final HolderLookup.Provider provider, @N
     @Override
     public void serialize(IFactoryController controller, ItemStorage input, RegistryFriendlyByteBuf packetBuffer)
     {
-        Utils.serializeCodecMess(packetBuffer, input.getItemStack());
+        // Some mods do weird things with banner patterns atm, so we add some special backup functionality here.
+        // We might have to extend this to other types of items in the future.
+        if (input.getItemStack().has(DataComponents.BANNER_PATTERNS))
+        {
+            boolean success = false;
+            try
+            {
+                Utils.serializeCodecMess(ItemStack.OPTIONAL_CODEC, packetBuffer.registryAccess(), input.getItemStack());
+                success = true;
+                Utils.serializeCodecMess(packetBuffer, input.getItemStack());
```

Comment:
this seems odd, why are we doing two serializations incase it works?

Reply:
weird logic structure overall, why the boolean at all? If the actual serialize to buffer call fails then it wont write an empty stack either?

Reply:
if the nbt one fails, the buf one will also fail, inversely if the nbt one passes the buf one should also pass. It's obviously not guaranteed, but pretty much the best we can do

Reply:
Could the empty stack serialisation not be moved into the catch () block instead?

Reply:
I didn't want to write into it, if the buf partially wrote into it.

Reply:
You can store the write index before writing, and then if failed reset to that, and wrote something new, overwriting the broken data

Reply:
^^ was about to say that too, save the writer index and reset to it